src/core/data.jl

```diff
@@ -3,348 +3,453 @@
+Clear damage indicator and replace with `damage=0`.
+"""
+function clear_damage_indicator!(data::Dict{String, <:Any})
+    pm_data = _PM.get_pm_data(data)
+    if _IM.ismultinetwork(pm_data)
+        for (nwid,nw) in pm_data["nw"]
+            _clear_damage_indicator!(nw)
         end
     else
-        _update_status!(pm_data, solution)
+        _clear_damage_indicator!(pm_data)
     end
 end
 
-function _update_status!(network::Dict{String,Any}, solution::Dict{String, Any})
-    for (comp_name, status_key) in _PM.pm_component_status
-        if haskey(solution, comp_name) && haskey(network, comp_name)
-            nw_comps = network[comp_name]
-            
-            for (i, sol_comp) in solution[comp_name]
-                nw_comp = nw_comps[i]
-                nw_comp[status_key] = sol_comp[status_key]
+
+""
+function _clear_damage_indicator!(network::Dict{String,<:Any})
+    for comp_name in restoration_comps
+        status_key = _PM.pm_component_status[comp_name]
+
+        for (i, comp) in get(network, comp_name, Dict())
+            if haskey(comp, "damaged")
+                comp["damaged"] = 0
             end
-        else
-            #TODO throw warning
         end
     end
 end
 
 
-#=
-"updates load and shunt demands, assumes these are continuous control parameters"
-function update_demand!(data::Dict{String, Any}, solution::Dict{String, Any})
-    @assert InfrastructureModels.ismultinetwork(data) == InfrastructureModels.ismultinetwork(solution)
+"""
+    clean_solution!(solution::Dict{String,<:Any})
 
-    if InfrastructureModels.ismultinetwork(solution)
-        for (i, nw_sol) in solution["nw"]
-            nw_data = data["nw"][i]
-            _update_demand!(nw_data, nw_sol)
+Replace variables with values of NaN or Nothing with 0 in solution dictionaries.
+"""
+function clean_solution!(solution::Dict{String,<:Any})
+    if _IM.ismultinetwork(solution["solution"])
+        for (nwid,nw) in solution["solution"]["nw"]
+            _clean_solution!(nw)
         end
     else
-        _update_demand!(data, solution)
+        _clean_solution!(solution["solution"])
     end
 end
 
-function _update_demand!(network::Dict{String,Any}, solution::Dict{String, Any})
-    if haskey(solution, "load") && haskey(network, "load")
-        nw_loads = network["load"]
-        for (i, sol_load) in solution["load"]
-            nw_load = nw_loads[i]
-            if !isnan(sol_load["pd"])
-                nw_load["pd"] = sol_load["pd"]
-            end
-            if !isnan(sol_load["qd"])
-                nw_load["qd"] = sol_load["qd"]
+
+""
+function _clean_solution!(data::Dict{String,<:Any})
+    for comp_type in keys(_PM.pm_component_status)
+        for (comp_id,comp) in get(data, comp_type, Dict())
+            for k in keys(comp)
+                if comp[k] === nothing
+                    comp[k] = 0
+                elseif typeof(comp[k])<:Number && isnan(comp[k])
+                    comp[k] = 0
+                end
```

Comment:
I am quite a bit confused by this code.  Checks like ` comp[k] === nothing` and `isnan(comp[k])` should generally be avoided.  Can you tell me more about how this is used?

Reply:
The primary check was for NaN when that occured in the solutions.  the nothing check does not account for when a key does not exist, I am not aware of any conditions where the solution dictionary would contain a nothing.

I removed the function from the tests for MRSP->ROP, ROP->redispatch and the tests still pass.

I removed the function from PMR.